programs/survival/item_holding_mobs_despawn_rule.sc

```diff
@@ -0,0 +1,24 @@
+// changes despawn rule for monster mobs. 
+// makes mobs who hold items despawnable as opposed to vanilla where they will never despawn.
+// mobs can only be made despawnable when they are given a custom name using nametags
+
+__command()->null;
+
+__config()->{
+    'stay_loaded' -> true;
+};
+
+__change_persistence() -> (
+    for ( entity_list('monster'),
+        if (_ ~ 'persistence' && _ ~ 'holds' && ! _ ~ 'custom_name',
+            modify(_, 'persistence', false);
+        )
+    )
+);
+
+global_counter = 0;
+
+__on_tick()->(
+    if (global_counter == 0,  schedule(0, '__change_persistence'));
+    global_counter = (global_counter + 1) % 20;

```

Comment:
You can use `if(! tick_time() % 20, ...)` instead

Reply:
Thanks for the suggestion. I will update it